Copy constructor and assignment operator warnings on Windows

On Windows only we get this warning:

```console
D:\a\srcML\srcML\src\parser\srcMLState.hpp(265,6): warning C5267: definition of implicit copy constructor for 'srcMLState' is deprecated because it has a user-provided destructor [D:\a\srcML\srcML\build\src\parser\parser.vcxproj]
  (compiling source file '../../../src/parser/DocstringPython.cpp')
      D:\a\srcML\srcML\src\parser\ModeStack.hpp(484,24):
      implicit copy constructor for 'srcMLState' first required here

D:\a\srcML\srcML\src\libsrcml\language_extension_registry.hpp(43,6): warning C5267: definition of implicit assignment operator for 'language_extension_registry' is deprecated because it has a user-provided destructor [D:\a\srcML\srcML\build\src\libsrcml\libsrcml.vcxproj]
  (compiling source file '../../../src/libsrcml/srcml_archive.cpp')
      D:\a\srcML\srcML\src\libsrcml\srcml_types.hpp(113,1):
      implicit assignment operator for 'language_extension_registry' first required here
      D:\a\srcML\srcML\src\libsrcml\srcml_types.hpp(113,1):
      This diagnostic occurred in the compiler generated function 'srcml_archive &srcml_archive::operator =(const srcml_archive &)'
      D:\a\srcML\srcML\src\libsrcml\srcml_archive.cpp(140,18):
      see reference to function 'srcml_archive &srcml_archive::operator =(const srcml_archive &)'
```

As it says, the implicit copy constructor and assignment operators are not created by default. In this case MSVC is pickier than GCC/Clang.

--- a/src/libsrcml/language_extension_registry.hpp
+++ b/src/libsrcml/language_extension_registry.hpp
@@ -42,6 +42,9 @@ public:
     language_extension_registry();
     ~language_extension_registry();
 
+    language_extension_registry(const language_extension_registry&) = default;
+    language_extension_registry& operator=(const language_extension_registry&) = default;
+
     bool register_user_ext(const char* ext, int language);
     bool register_user_ext(const char* ext, const char* language);
 

--- a/src/parser/srcMLState.hpp
+++ b/src/parser/srcMLState.hpp
@@ -40,6 +40,12 @@ public:
         : flags(mode), flags_prev(prevmode), flags_all(transmode | mode), parencount(0), curlycount(0), typecount(0)
     {}
 
+    // copy constructor
+    srcMLState(const srcMLState&) = default;
+
+    // copy assignment
+    srcMLState& operator=(const srcMLState&) = default;
+
     /**
      * size
      *

